Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -27,12 +33,137 @@ def _getBasename(filename):
+            "sample_container": ws.run()['SampleContainer'].lastValue().replace(" ", ""),
+            "algorithm_used": alg_name,
+        }.items()
+    ]
+
+    cache_path, signature = mantid.simpleapi.CreateCacheFilename(
+        Prefix=prefix,
+        OtherProperties=property_string,
+        CacheDir=cache_dir,
+    )
+
+    return cache_path, signature
+
+
+def _getCachedData(absName, abs_method, sha1, cache_file_name):
+    """
+    With a given absorption workspace name, check both memory and disk to see if we can locate the cache
+    """
+    wsn_as = ""  # absorption workspace sample
+    wsn_ac = ""  # absorption workspace container
+    found_as = False
+    found_ac = False
+
+    # step_0: depending on the abs_method, suffix will be different
+    if abs_method == "SampleOnly":
+        wsn_as = f"{absName}_ass"
+        found_ac = True
+    elif abs_method == "SampleAndContainer":
+        wsn_as = f"{absName}_ass"
+        wsn_ac = f"{absName}_acc"
+    elif abs_method == "FullPaalmanPings":
+        wsn_as = f"{absName}_assc"
+        wsn_ac = f"{absName}_ac"
+    else:
+        raise RuntimeWarning("Unrecognized absorption correction method '{}'".format(abs_method))
+
+    # step_1: check memory to see if the ws is already there
+    # -- check SHA1
+    # NOTE: The new attribute absSHA1 should be there.
+    if mtd.doesExist(wsn_as):
+        found_as = mtd[wsn_as].run()["absSHA1"].value == sha1
+    if mtd.doesExist(wsn_ac):
+        found_ac = mtd[wsn_ac].run()["absSHA1"].value == sha1
+
+    # step_2: try to load from cache_file provided
+    if (not found_as) or (not found_ac):
+        # disk -> memory
+        if os.path.exists(cache_file_name):
+            # LoadNexus(Filename=cache_file_name, OutputWorkspace=wsn_as)
+            # if mtd[wsn_as].id() == "WorkspaceGroup":
+            #     UnGroupWorkspace(InputWorkspace=wsn_as)
+            wsntmp = f"{wsn_as}_wsg"
+            Load(Filename=cache_file_name, OutputWorkspace=wsntmp)
+            wstype = mtd[wsntmp].id()
+            if wstype == "Workspace2D":
+                RenameWorkspace(InputWorkspace=wsntmp, OutputWorkspace=wsn_as)
+            elif wstype == "WorkspaceGroup":
+                # there should be exactly two workspaces inside, ungroup them will
+                # restore them with the orignal name (wsn_as, wsn_ac)
+                UnGroupWorkspace(InputWorkspace=wsntmp)
+            else:
+                raise ValueError(f"Unsupported caching type: {wstype}")
+
+        # now check the memory again
+        # -- wsn_as exist for all three methods
+        # NOTE: The new attribute absSHA1 should be there.
+        if mtd.doesExist(wsn_as):
+            found_as = mtd[wsn_as].run()["absSHA1"].value == sha1
+        if mtd.doesExist(wsn_ac):
+            found_ac = mtd[wsn_ac].run()["absSHA1"].value == sha1
+
+    # step_3: if we did not find the cache, set both to None
```

Comment:
maybe "assign an empty string to both"?

Reply:
comments updated to match the actual implementation.